packages/react-native/src/storage/FileChunkSink.ts

```diff
@@ -0,0 +1,83 @@
+import type { ChunkSink } from './Chunkifier';
+import type { FileSystem } from './FileSystem';
+import type { FileWritableStream } from './StreamWriter';
+
+interface FileChunkSinkOptions {
+    /**
+     * Maximum number of files.
+     */
+    readonly maxFiles: number;
+
+    /**
+     * Full path to the chunk file.
+     */
+    readonly file: (n: number) => string;
+
+    /**
+     * File system to use.
+     */
+    readonly fs: FileSystem;
+}
+
+/**
+ * Chunk sink which writes data to disk.
+ *
+ * Each time a new chunk is created, a new stream is created with path provided from options.
+ */
+export class FileChunkSink {
+    private readonly _streamTracker: LimitedFifo<FileWritableStream>;
+
+    /**
+     * Returns all files that have been written to and are not deleted.
+     */
+    public get files() {
+        return this._streamTracker.elements;
+    }
+
+    constructor(private readonly _options: FileChunkSinkOptions) {
+        // Track files using a FIFO queue
+        this._streamTracker = limitedFifo<FileWritableStream>(_options.maxFiles, async (stream) => {
+            await stream.close().finally(() => _options.fs.unlink(stream.path));
+        });
+    }
+
+    /**
+     * Returns `ChunkSink`. Pass this to `chunkifier`.
+     */
+    public getSink(): ChunkSink<string, FileWritableStream> {
+        return (n) => {
+            const stream = this.createStream(n);
+            this._streamTracker.push(stream);
+            return stream;
+        };
+    }
+
+    private createStream(n: number) {
+        const path = this._options.file(n);
+        // TODO: What to do if this returns undefined?
+        return this._options.fs.createWriteStream(path);
```

Comment:
we can't perform any stream operation if it returns undefined.

Reply:
yeah, this is outdated, forgot to remove this.